---
title: "07_PCA"
format: html
editor: visual
---

```{r}
# output: FALSE
library(tidyverse)
library(ggrepel)
library(broom)
```

```{r}
# Load augmented data for PCA analysis
expression_data <- read_tsv(gzfile('../data/augmented_meta_gene_table.tsv.gz'))
```

```{r}
all_cols <- expression_data |> 
  colnames()
```

```{r}
# Exclude data columns that will not be used for PCA
pca_data <- expression_data |> 
  dplyr::select(-one_of(c("log2_Lesion",
                          "log2_Perilesion",
                          "GENENAME",
                          "CHR",
                          "log2_expr_change",
                          "expr_diff",
                          "Sample",
                          "SYMBOL"))) |> 
  # Convert age and BMI vars into categories
  mutate(Age = cut_interval(Age, length = 10)) |> 
  mutate(BMI = case_when(BMI < 18.5 ~ 'Underweight',
                         18.5 <= BMI & BMI < 25 ~ 'Healthy weight',
                         25 <= BMI & BMI < 30 ~ 'Overweight',
                         BMI >= 30 ~ 'Obese'))
  
```

```{r}
# Wrangle the dataset into a wide format, with Tissue as a single column
pca_data <- pca_data |> 
  pivot_longer(c("Lesion","Perilesion"),
               names_to = "Tissue",
               values_to = "expr") |> 
  pivot_wider(id_cols = c("Gender",
                          "Race",
                          "Smoker",
                          "Age",
                          "BMI",
                          #"SYMBOL",
                          "Tissue"),
              names_from = "gene",
              values_from = "expr",
              values_fn = mean)
```

```{r}
# perform PCA on gene expression data
pca_fit <- pca_data |> 
  dplyr::select(starts_with('EN')) |> 
  scale() |> 
  prcomp() 
```

```{r}
# PCA score plot
PCA_score <- pca_fit |> 
  augment(pca_data) |> 
  ggplot(aes(.fittedPC1, 
             .fittedPC2, 
             color = Tissue,
             #label = Race
             )) + 
  geom_point(size = 2.5) +
  labs(title = 'PCA Score Plot of RNA-seq Data by Tissue Type',
       x = 'PC 1',
       y = 'PC 2',
       color = 'Tissue') +
  # Adds sample labels
  #geom_text_repel(size = 2.5,
  #                min.segment.length = 0,
  #                segment.linetype = 4) + 
  theme_minimal() +
  theme(plot.title = 
          element_text(hjust = 0.5,
                       vjust = 2,
                       face = 'bold'),
        panel.grid.minor = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14)
        ) 

  ggsave("PCA_score.png", PCA_score, path = "../pictures") 

  PCA_score
```

```{r}
# Figure for explained variance
PCA_var <- pca_fit |>
  tidy(matrix = "eigenvalues") |>
  filter(PC <= 10) |>
  ggplot(aes(PC, percent)) +
  geom_col() +
  scale_x_continuous(breaks = 1:10) +
  scale_y_continuous(
    labels = scales::percent_format(),
    limits = c(0,1)
    #expand = expansion(mult = c(0.12, 0.3))
  ) +
  geom_point(aes(
      x = PC,
      y = cumulative
    ),
    color = 'black') +
  geom_line(aes(
      x = PC,
      y = cumulative
    ),
    linetype = 'dashed',
    color = 'black')+
  ylab('Variance Explained (%)')+
  xlab('Principal Component')+
  ggtitle("Individual and Cumulative variance explained by first 10 PC's")+
  theme_minimal()+
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 14))


  ggsave("PCA_var.png", PCA_var, path = "../pictures") 
  
  PCA_var
```
