---
title: "Read_count_analysis"
format: html
editor: visual
---

```{r}
library(tidyverse)
```

```{r}
# Calculate the buffer size in bytes (19.7 MB) size of meta_gene_table
buffer_size_in_bytes <- 19.7 * 1048576

# Set the environment variable for vroom
Sys.setenv(VROOM_CONNECTION_SIZE = buffer_size_in_bytes)
```

```{r}
#Load in our raw data
raw_data <- read_tsv(gzfile('../data/meta_gene_table.tsv.gz'))
```

## Read counts for each sample

A comparison of read counts between our significant and all of our genes can give us an idea about the differing sequencing depths, which can potentially color our analysis. Comparing the number of reads of different samples also gives us an idea about the inherent bias that might be present in our data - expression signals from samples having a greater sequencing depth might dominate.

```{r}
#Calculate total number of genes in raw data
raw_gene_count <- raw_data |> 
  pivot_longer(gene_cols, 
               names_to = 'Gene', 
               values_to = 'Expression') |> 
  group_by(Gene) |> 
  summarise()
```

```{r fig.height = 6.5, fig.width = 6.5 }
#Extract the gene columns
gene_cols <- 
  raw_data |> 
  dplyr::select(starts_with('ENSG')) |> 
  colnames()
  
#Factorizing the samples
Sample_count <- 
  raw_data |> 
  pivot_longer(gene_cols, 
               names_to = 'Gene', 
               values_to = 'Expression') |> 
  group_by(Sample) |> 
  summarise(Read_count = sum(Expression)) |> 
  mutate(Sample = factor(Sample, 
                         levels = c("1L", "1N",
                                    "2L", "2N",
                                    "3L", "3N",
                                    "4L", "4N",
                                    "5L", "5N",
                                    "6L", "6N",
                                    "7L", "7N",
                                    "8L", "8N",
                                    "9L", "9N",
                                    "10L", "10N", 
                                    "11L", "11N", 
                                    "12L", "12N", 
                                    "13L", "13N", 
                                    "14L", "14N",
                                    "15L", "15N",
                                    "16L", "16N",
                                    "17L", "17N",
                                    "18L", "18N",
                                    "19L", "19N",
                                    "20L", "20N",
                                    "21L", "21N",
                                    "22L", "22N")
         ))


reads <- Sample_count |> 
  ggplot(aes(x = fct_rev(Sample), y = Read_count, fill = "cyan")) +
  geom_bar(stat = 'identity') +
  coord_flip() +
  labs(title = 'Number of Raw Reads in Each Sample',
       x = 'Sample',
       y = 'Number of reads') +
  theme_minimal() +
  theme(plot.title = element_text(vjust = 2,
                                  hjust = 0.5,
                                  face = 'bold'),
        panel.border = element_rect(colour = "black", 
                                    fill = 'transparent',
                                    size = 0.5),
        legend.position = 'none')

reads
ggsave(filename = "Read_count_raw.png", reads, path = "../pictures", bg = "white")
```

```{r}
#Total number of reads for all genes in raw data
Total_raw_reads <- Sample_count |> 
  summarise(Total= sum(Read_count))
```

```{r}
#Load in our significant genes
augmented_data <- read_tsv('../data/augmented_meta_gene_table.tsv.gz')
```

```{r fig.height = 6.5, fig.width = 6.5}

#Count the number of significant genes
sig_gene_count <- augmented_data |> 
  group_by(gene) |> 
  summarise()

#Factorizing the samples
augmented_counts <- 
  augmented_data |> 
  pivot_longer(c(Lesion, Perilesion), 
               names_to = 'Tissue_type', 
               values_to = 'expr') |> 
  mutate(Sample = ifelse(Tissue_type == 'Lesion', 
                        str_c(Sample, 'L'), 
                        str_c(Sample, 'N')
                        )) |> 
  group_by(Sample) |> 
  summarise(Read_count = sum(expr)) |> 
  mutate(Sample = factor(Sample, 
                         levels = c("1L", "1N",
                                    "2L", "2N",
                                    "3L", "3N",
                                    "4L", "4N",
                                    "5L", "5N",
                                    "6L", "6N",
                                    "7L", "7N",
                                    "8L", "8N",
                                    "9L", "9N",
                                    "10L", "10N", 
                                    "11L", "11N", 
                                    "12L", "12N", 
                                    "13L", "13N", 
                                    "14L", "14N",
                                    "15L", "15N",
                                    "16L", "16N",
                                    "17L", "17N",
                                    "18L", "18N",
                                    "19L", "19N",
                                    "20L", "20N",
                                    "21L", "21N",
                                    "22L", "22N")))
  
reads_sig <- augmented_counts |> 
  ggplot(aes(x = fct_rev(Sample), y = Read_count, fill = "cyan")) +
  geom_bar(stat = 'identity') +
  coord_flip() +
  labs(title = 'Number of Reads of Significant Genes in Each Sample',
       x = 'Sample',
       y = 'Number of reads') +
  theme_minimal() +
  theme(plot.title = element_text(vjust = 2,
                                  hjust = 0.5,
                                  face = 'bold'),
        panel.border = element_rect(colour = "black", 
                                    fill = 'transparent',
                                    size = 0.5),
        legend.position = 'none')

reads_sig
ggsave("Read_count_significant.png", reads_sig, path = "../pictures", bg = "white")

```

```{r}
#Counting total number of reads of our significant genes
Total_sig_reads <- augmented_counts |> 
  summarise(Total = sum(Read_count))
```
