---
title: "Presentation"
format: 
  revealjs: 
    theme: dark
    slide-number: true
    show-slide-number: all
    embed-resources: true
editor: visual
---

# Presentation Group 5


## Introduction

Data from undefined A Hidradenitis Suppurativa Molecular Disease Signature Derived from Patient Samples by High-Throughput RNA Sequencing available on GEO RNASeq results for 40 samples.

## Trace the Flow of the Project


![](../pictures/flowchart_cut.png)
## Load

## Data Cleaning

- Start with <span style="color:red;">58051</span> genes in the pool
  1. Remove gene if 0 expression for all patients (<span style="color:orange;">-6098</span>)
  2. Remove gene if expression linear model coefficient not significant between diseased/healthy tissue (p>0.05) (<span style="color:orange;">-51437</span>)
  3. Remove gene if expression change for all patients < 3-fold between the diseased/healthy tissue groups ( <span style="color:orange;">-29</span>)
  
- <span style="color:green;">487</span> genes advance to analysis rounds



## Materials and Methods

We read the data from GSE151243

``` {.r code-line-numbers="3"}
# Loading the data
path_gene = "../data/_raw/GSE151243_Raw_gene_counts_matrix.txt"
path_meta = "../data/_raw/GSE151243_series_matrix.txt"
gene_table <- read.table(path_gene, header = TRUE)
meta_table <- read.table(path_meta, skip = 33, fill = TRUE, header = TRUE)
```

## Results

How to add a picture from the pictures folder

![PCA plot for augmented data](../pictures/PCA_score.png)

## Discussion

When you click the **Render** button a presentation will be generated that includes both content and the output of embedded code. You can embed code like this:

```{r}
1 + 1
```
