---
title: "Presentation Group 5"
subtitle: "Emmanuel, Adomas, Subhayan and Chuang"
format: 
  revealjs: 
    theme: dark
    slide-number: true
    show-slide-number: all
    embed-resources: true
    fontsize: "20pt"

editor: visual
---

## Introduction

-   Hidradenitis suppurativa (HS) - inflammatory skin disease linked to immune dysregulation and abnormalities in follicular structure and function.

-   Data - Lesional and non-lesional skin samples from 20 patients

-   Study objective - Find an expression-based disease signature of HS by observing affected and unaffected skin samples.

-   Methods - Differential expression and pathway enrichment analyses.

The dataset used for the study isn't particularly diverse - for instance, only 3 males were sampled, all of whom were either active or ex-smokers.

## Trace the Flow of the Project

![](../pictures/flowchart.bmp){fig-align="center"}

## Load

We read the data from GSE151243

``` {.r code-line-numbers="3"}
# Loading the data
path_gene = "../data/_raw/GSE151243_Raw_gene_counts_matrix.txt"
path_meta = "../data/_raw/GSE151243_series_matrix.txt"
gene_table <- read.table(path_gene, header = TRUE)
meta_table <- read.table(path_meta, skip = 33, fill = TRUE, header = TRUE)
```

## Data Cleaning

-   Start with [58051]{style="color:red;"} genes in the pool
    1.  Remove gene if 0 expression for all patients ([-6098]{style="color:orange;"})
    2.  Remove gene if expression linear model coefficient not significant between diseased/healthy tissue (p\>0.05) ([-51437]{style="color:orange;"})
    3.  Remove gene if expression change for all patients \< 3-fold between the diseased/healthy tissue groups ( [-29]{style="color:orange;"})
-   [487]{style="color:green;"} genes advance to analysis rounds

## Read Counts

::: columns
::: {.column width="50%"}
![](../pictures/Read_count_raw.png)
:::

::: {.column width="50%"}
![](../pictures/Read_count_significant.png)
:::
:::

## Chromosome Distribution of Upregulated genes

::: columns
::: {.column width="50%"}
![](../pictures/chromosomal_dist_upreg.png){fig-align="center"}
:::

::: {.column width="50%"}
![](../pictures/chromosomal_dist_downreg.png){fig-align="center"}
:::
:::

## Results

## Principal Component Analysis

::: columns
::: {.column width="50%"}
![Score Plot](../pictures/PCA_score.png)
:::

::: {.column width="50%"}
![Variance Explained](../pictures/PCA_var.png)
:::
:::

## Most Upregulated and Downregulated Genes

![](../pictures/Heatmap_overview.png){fig-align="center"}

## Functional Assessment of Upregulated and Downregulated Genes

-   Upregulated genes:
    -   **NTRK1:** Neuronal survival and Signal Transduction\\n(MAPK, PI3K-Akt, PLC-$\gamma$)
    -   **POU2AF1:** B-Cell immune responds
    -   **SIGLEC7:** Inhibition of NK-Cell responds
-   Downregulated genes:
    -   **KRT77:** Kreatin, Intermediate filament protein
    -   **CHRM1:** Muscarinic receptor
    -   **PHYHIP:** Fatty acid metabolism and perioxisomal function

## Enhanced Activation of Inflammatory, Innate and Adaptive immune pathway
::: columns
::: {.column width="50%"}
![](../pictures/GO_upreg_top20.png)
:::

::: {.column width="50%"}
![](../pictures/GO_downreg_overview.png){fig-align="center"}
:::
:::

## Discussion

-   Observations by Freudenberg et al.
    -   Gene Upregulation
        -   MMP1, C-X-C, SERPINB4, and S100.
    -   Enhanced Immune Responses
        -   Upregulation in both adaptive and innate immune system functions.
    -   Reduced Keratinization and Intermediate Filament Activity
-   Potential Causes for Variance in Results
    -   Model Selection
        -   lm vs glm, Empirical Bayes
    -   Adjusted P-value Cutoff Choices
    -   Gene Selection Criteria
        -   Focus only on protein-encoding genes



