---
title: "Presentation Group 5"
subtitle: "Emmanuel, Adomas, Subhayan, Chuang"
format: 
  revealjs: 
    theme: dark
    slide-number: true
    show-slide-number: all
    embed-resources: true
    fontsize: "20pt"

editor: visual
---

## Introduction

Data from undefined A Hidradenitis Suppurativa Molecular Disease Signature Derived from Patient Samples by High-Throughput RNA Sequencing available on GEO RNASeq results for 40 samples.

## Trace the Flow of the Project

![](../pictures/flowchart_cut.png) 

## Load

## Data Cleaning


- Start with <span style="color:red;">58051</span> genes in the pool
  1. Remove gene if 0 expression for all patients (<span style="color:orange;">-6098</span>)
  2. Remove gene if expression linear model coefficient not significant between diseased/healthy tissue (p>0.05) (<span style="color:orange;">-51437</span>)
  3. Remove gene if expression change for all patients < 3-fold between the diseased/healthy tissue groups ( <span style="color:orange;">-29</span>)
  
- <span style="color:green;">487</span> genes advance to analysis rounds


## Materials and Methods

We read the data from GSE151243

``` {.r code-line-numbers="3"}
# Loading the data
path_gene = "../data/_raw/GSE151243_Raw_gene_counts_matrix.txt"
path_meta = "../data/_raw/GSE151243_series_matrix.txt"
gene_table <- read.table(path_gene, header = TRUE)
meta_table <- read.table(path_meta, skip = 33, fill = TRUE, header = TRUE)
```

## Results

## Principal Component Analysis

:::: {.columns}

::: {.column width="50%"}
![Score Plot](../pictures/PCA_score.png)
:::

::: {.column width="50%"}


![Variance Explained](../pictures/PCA_var.png)

:::
::::

## Gene Ontology Analysis

## Discussion

-   Observations by Freudenberg et al.
    -   Gene Upregulation
        -   MMP1, C-X-C, SERPINB4, and S100.
    -   Enhanced Immune Responses
        -   Upregulation in both adaptive and innate immune system functions.
    -   Reduced Keratinization and Intermediate Filament Activity

-   Potential Causes for Variance in Results
    -   Model Selection
        -   lm vs glm, Empirical Bayes
    -   Adjusted P-value Cutoff Choices
    -   Gene Selection Criteria
        -   Focus only on protein-encoding genes
