---
title: "Presentation Group 5"
subtitle: "Emmanuel, Adomas, Subhayan and Chuang"
format: 
  revealjs: 
    theme: dark
    slide-number: true
    show-slide-number: all
    embed-resources: true
    fontsize: "20pt"

editor: visual
---

## Introduction

Hidradenitis suppurativa (HS) is a common, debilitating inflammatory skin disease linked to immune dysregulation and abnormalities in follicular structure and function. The study we looked at tries to derive an expression-based HS disease signature by observing affected and unaffected skin from 20 subjects and characterize a transcriptomic profile through RNASeq. Differential expression and pathway enrichment analyses were then performed, and the results were compared to previously reported datasets.

\
We obtained the RNASeq data used for the study - A Hidradenitis Suppurativa Molecular Disease Signature Derived from Patient Samples by High-Throughput RNA Sequencing, available on GEO RNASeq results for 40 samples - and tried to replicate the differential expression profiling and pathway enrichment analyses.

## Trace the Flow of the Project

![](../pictures/flowchart_cut.png)

## Load

We read the data from GSE151243

``` {.r code-line-numbers="3"}
# Loading the data
path_gene = "../data/_raw/GSE151243_Raw_gene_counts_matrix.txt"
path_meta = "../data/_raw/GSE151243_series_matrix.txt"
gene_table <- read.table(path_gene, header = TRUE)
meta_table <- read.table(path_meta, skip = 33, fill = TRUE, header = TRUE)
```

## Data Cleaning

-   Start with 58051 genes in the pool
    -   Remove gene if 0 expression for all patients (-6098)
    -   Remove gene if expression linear model coefficient not significant (p\>0.05) (-51437)
    -   Remove gene if expression change \< 3-fold between the diseased tissue groups (-29)
-   487 Genes remain for analysis

## Materials and Methods

## Results

How to add a picture from the pictures folder

![PCA plot for augmented data](../pictures/PCA_score.png)

## Most Upregulated and Downregulated Genes

![](../pictures/Heatmap_overview.png){fig-align="center"}

## Functional Assessment of Upregulated and Downregulated Genes

-   Upregulated genes:
    -   **NTRK1:** Neuronal survival and Signal Transduction\\n(MAPK, PI3K-Akt, PLC-$\gamma$)
    -   **POU2AF1:** B-Cell immune responds
    -   **SIGLEC7:** Inhibition of NK-Cell responds
-   Downregulated genes:
    -   **KRT77:** Kreatin, Intermediate filament protein
    -   **CHRM1:** Muscarinic receptor
    -   **PHYHIP:** Fatty acid metabolism and perioxisomal function

## Enhanced Activation of Inflammatory, Innate and Adaptive immune pathway

![](../pictures/GO_upreg_top20.png){fig-align="center"}

## Reduced Activity in Hair, Skin and Neural Cell Division Pathways

![](../pictures/GO_downreg_overview.png){fig-align="center"}

## Discussion

-   Observations by Freudenberg et al.
    -   Gene Upregulation
        -   MMP1, C-X-C, SERPINB4, and S100.
    -   Enhanced Immune Responses
        -   Upregulation in both adaptive and innate immune system functions.
    -   Reduced Keratinization and Intermediate Filament Activity
-   Potential Causes for Variance in Results
    -   Model Selection
        -   lm vs glm, Empirical Bayes
    -   Adjusted P-value Cutoff Choices
    -   Gene Selection Criteria
        -   Focus only on protein-encoding genes

## Chromosome Distribution of Upregulated genes

![](images/chromosomal_dist_upreg-01.png){fig-align="center" width="500"}

## Chromosome Distribution of Downregulated genes

![](images/chromosomal_dist_downreg-01.png){fig-align="center" width="500"}

## Read Counts are higher for Raw genes

![](images/Read_count_raw-01.png){fig-align="center" width="500"}

## Some nonlesional samples have higher read counts of Significant genes

![](images/Read_count_significant-02.png){fig-align="center" width="500"}
